### PR TITLE
UI refactor

### PR DESCRIPTION
**Pull request by tchassijordan**
### Major Changes
- Refactored the UI of the Sidebar Component. The old UI was too plain, UI design fundamentals weren't respected. So I refactored all those out.
- Broke down the components of the app into separate reusable components with each having a single responsibility.
- Refactored source code

###Preview Change:
 
Screenshot from 2022-04-23 02-39-26

### CODE REVIEW

File: components/NavbarItem.tsx [hunk 1]
```diff
@@ -0,0 +1,24 @@
+import Link from "next/link";
+import { ReactElement, ReactNode } from "react";
+
+type Params = {
+  children?: ReactNode;
+  to: string;
+  cls?: string;
+};
+
+export default function NavbarItem({
```
Comment (fdrissi) [inline components/NavbarItem.tsx]: SidbarItem will be a better naming

File: components/Sidebar.tsx [hunk 2]
```diff
@@ -1,11 +1,14 @@
...
   return (
-    <div className="w-1/5 bg-stone-900 text-white border-r-2">
-      <ul className="list-none flex flex-col items-center gap-3 pt-5">
-        <li className="hover:text-stone-400 cursor-pointer border-b-2 border-stone-400">Home</li>
-        <li className="hover:text-stone-400 cursor-pointer border-b-2 border-stone-400">About</li>
-        <li className="hover:text-stone-400 cursor-pointer border-b-2 border-stone-400">Contact</li>
-      </ul>
+    <div className="w-64 bg-[#222628] text-white flex flex-col border-[1px] border-[#a5a19950]">
+      <h1 className="px-5 py-4 mb-3 font-bold text-lg tracking-wide cursor-pointer border-b-[1px] border-[#a5a19950]">Quick Links</h1>
+      {children}
```
Comment (fdrissi) [inline components/Sidebar.tsx]: Move the title to the props, make it optional, and set the default value to `Quick Links`

File: components/Sidebar.tsx [hunk 3]
```diff
@@ -1,11 +1,14 @@
...
+
+const Sidebar = ({ children }: Params):ReactElement => {
   return (
-    <div className="w-1/5 bg-stone-900 text-white border-r-2">
-      <ul className="list-none flex flex-col items-center gap-3 pt-5">
-        <li className="hover:text-stone-400 cursor-pointer border-b-2 border-stone-400">Home</li>
-        <li className="hover:text-stone-400 cursor-pointer border-b-2 border-stone-400">About</li>
-        <li className="hover:text-stone-400 cursor-pointer border-b-2 border-stone-400">Contact</li>
-      </ul>
+    <div className="w-64 bg-[#222628] text-white flex flex-col border-[1px] border-[#a5a19950]">
```
Comment (fdrissi) [inline components/Sidebar.tsx]: border-[1px] === border.
bg-[#222628] this maybe will be your primary color, and this border-[#a5a19950] maybe your primary border color, so add them to tw theme.

File: components/Table.tsx [hunk 4]
```diff
@@ -0,0 +1,51 @@
+import React, { ReactElement, useState } from "react";
+import { AgGridReact } from "ag-grid-react";
+import data from "../utils/data";
+import "ag-grid-enterprise";
+import "ag-grid-community/dist/styles/ag-grid.css";
+import "ag-grid-community/dist/styles/ag-theme-alpine-dark.css";
+
+export default function Table():ReactElement {
+  //define the column header for our table in a local state
+  const [columnDefs] = useState([
```
Comment (fdrissi) [inline components/Table.tsx]: This is not a reusable component.

File: pages/index.tsx [hunk 5]
```diff
@@ -1,65 +1,24 @@
...
-            animateRows={true}
-            rowGroupPanelShow={"always"}
-          ></AgGridReact>
-        </div>
+      <div className="w-full h-full flex bg-[#151718] gap-1 p-1 font-primary">
+        <Sidebar>
+          <NavbarItem to={"/"}>country</NavbarItem>
+          <NavbarItem to={"/"} cls="opacity-60">sectors</NavbarItem>
+          <NavbarItem to={"/"} cls="opacity-60">signed amount</NavbarItem>
+          <NavbarItem to={"/"} cls="opacity-60">date</NavbarItem>
```
Comment (fdrissi) [inline pages/index.tsx]: move these links to an array, and handle the active link style dynamically from inside `NavbarItem` component

**Review by fdrissi — COMMENTED** [review]
This is much better, you made good progress, good job.
don't forget to work also on commit messages

**Comment by tchassijordan** [comment]
@fdrissi Thanks so much for the review. 
I have improved the code base following your comments.

**Pull request merged**